### PR TITLE
Improved interpolation search level

### PR DESCRIPTION
**Pull request by NaderMahdi**
Added Chroma blind interpolation search at fast loop and set it to be used by encoder mode 0.

**Speed gain in context of M0 is about 18%  and about 0% loss in bdrate**

// Interpolation search Level                                                            Settings
----------------------------------------------------------------------------------
// IT_SEARCH_FAST_LOOP_UV_BLIND                                              Chroma blind interpolation search at fast loop

### CODE REVIEW

**Comment by EwoutH** [comment]
Saw a 8,2% performance improvement (0,428 --> 0,463 fps) on a dual Xeon system (#98).

- **Video:** Morocco (1080p 8-bit 4:2:0)
- **Settings:** `-enc-mode 0 -n 120`
- **System:** 2x Intel Xeon E5-2680v4  (2x 14c/28t, AVX2), 128GB DDR4 2133MHz, Linux Mint

File: Source/Lib/Codec/EbCodingLoop.c [hunk 1]
```diff
@@ -2359,7 +2359,13 @@ EbErrorType Av1QpModulationLcu(
         if (type == INTRA_MODE) {
 
 
+            
+#if OIS_BASED_INTRA
+            ois_sb_results_t        *ois_sb_results_ptr = picture_control_set_ptr->parent_pcs_ptr->ois_sb_results[sb_index];
+            ois_candidate_t *OisCuPtr = ois_sb_results_ptr->sorted_ois_candidate[from_1101_to_85[cu_index]];
```
Comment (joelsole) [inline Source/Lib/Codec/EbCodingLoop.c]: OisCuPtr => snake_case

File: Source/Lib/Codec/EbIntraPrediction.h [hunk 2]
```diff
@@ -368,6 +368,26 @@ extern "C" {
...
         IntraReferenceSamplesOpenLoop_t  *context_ptr);
 
+#if OIS_BASED_INTRA
+    extern EbErrorType update_neighbor_samples_array_open_loop(
+        uint8_t                           *above_ref,
+        uint8_t                            *left_ref,
+        EbPictureBufferDesc_t              *input_ptr,
+        uint32_t                            stride,
+        uint32_t                            srcOriginX,
+        uint32_t                            srcOriginY,
```
Comment (joelsole) [inline Source/Lib/Codec/EbIntraPrediction.h]: snake_case

File: Source/Lib/Codec/EbProductCodingLoop.c [hunk 3]
```diff
@@ -2656,6 +2684,19 @@ EbBool allowed_ns_cu(
...
     }
+
+#if NSQ_OPTIMASATION
+    if (picture_control_set_ptr->slice_type == !I_SLICE) {
+        if (context_ptr->blk_geom->shape != PART_N) {
+            ret = 0;
+            for (int i = 0; i < picture_control_set_ptr->parent_pcs_ptr->nsq_max_shapes_md; i++) {
+                if (context_ptr->blk_geom->shape == context_ptr->nsq_table[i]) {
+                    ret = 1;
+                }
```
Comment (joelsole) [inline Source/Lib/Codec/EbProductCodingLoop.c]: from style.md: braces are only necessary for multi-line code blocks or multi-line condition statements;

**Comment by hassount** [comment]
To be merged as part of #185

**Pull request closed without merging**